### PR TITLE
Added vulnerabilities tab in organization

### PR DESCRIPTION
**Pull request by Cheshiriks**
Снимок

### CODE REVIEW

**Review by nulls — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt [hunk 1]
```diff
@@ -130,6 +130,20 @@ class VulnerabilityController(
...
     }.switchIfEmptyToNotFound()
 
+    @GetMapping("/by-organization-and-status")
+    @Operation(
+        method = "GET",
+        summary = "Get list of vulnerabilities by organization name.",
+        description = "Get list of vulnerabilities by organization name.",
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully fetched list of vulnerabilities by organization name")
+    fun getVulnerabilityBorganization(
```
Comment (sanyavertolet) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt:140]:
```suggestion
    fun getVulnerabilityByOrganization(
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+val organizationVulnerabilitiesMenu = organizationVulnerabilitiesMenu()
+
+@Suppress("MAGIC_NUMBER", "TYPE_ALIAS")
+private val vulnerabilityTable: FC<TableProps<VulnerabilityDto>> = tableComponent(
+    columns = {
+        columns {
+            column(id = "name", header = "Name", { this.name }) { cellContext ->
+                Fragment.create {
+                    ReactHTML.td {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt:34]:
```suggestion
                    td {
```

one day...

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * OrganizationVulnerabilitiesMenu component props
+ */
+external interface OrganizationVulnerabilitiesMenuProps : Props {
+    /**
+     * Current organization name
+     */
+    var organizationName: String
+}
+
+private fun organizationVulnerabilitiesMenu() = FC<OrganizationVulnerabilitiesMenuProps> { props ->
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt:73]:
```suggestion
val organizationVulnerabilitiesMenu = FC<OrganizationVulnerabilitiesMenuProps> { props ->
```

there is no need to create fabric method

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt [hunk 4]
```diff
@@ -64,16 +64,7 @@ private val vulnerabilityTable: FC<TableProps<VulnerabilityDto>> = tableComponen
...
- */
-external interface OrganizationVulnerabilitiesMenuProps : Props {
-    /**
-     * Current organization name
-     */
-    var organizationName: String
-}
-
+@Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
 val organizationVulnerabilitiesMenu = FC<OrganizationVulnerabilitiesMenuProps> { props ->
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt:68]:
```suggestion
val organizationVulnerabilitiesMenu: FC<OrganizationVulnerabilitiesMenuProps> = FC { props ->
```

**Review by akuleshov7 — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt (hunk 4 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt]: Fixed

**Comment by akuleshov7** [comment]
can you please also rename column TESTS to BENCHMARKS

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt (hunk 4 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationVulnerabilitiesMenu.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt [hunk 5]
```diff
@@ -228,7 +228,7 @@ class OrganizationView : AbstractView<OrganizationProps, OrganizationViewState>(
         when (state.selectedMenu) {
             OrganizationMenuBar.INFO -> renderInfo()
             OrganizationMenuBar.TOOLS -> renderTools()
-            OrganizationMenuBar.TESTS -> renderTests()
+            OrganizationMenuBar.BENCHMARKS -> renderTests()
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt:231]: To my mind `renderTests` should also be renamed

Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt:231]: Fixed

Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt:231]: see no changes

Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt]: Saw in https://github.com/saveourtool/save-cloud/pull/2314/commits/46dfdbd5134e5ce6417e0d90cd776ca898dd1aa8

**Pull request merged**